In [1]:
import pandas as pd

In [4]:
df = pd.read_csv('C:\\Users\\Aneta\\Documents\\Studia\\pum\\winequality_white.csv', sep=';')
df.head(7)

fixed acidity  volatile acidity  citric acid  residual sugar chlorides  \
0            7.0              0.27         0.36            20.7     0.045   
1            6.3              0.30         0.34             1.6     0.049   
2            8.1              0.28         0.40             6.9      0.05   
3            7.2              0.23         0.32             8.5     0.058   
4            7.2              0.23         0.32             8.5     0.058   
5            8.1              0.28         0.40             6.9      0.05   
6            6.2              0.32         0.16             7.0     0.045   

  free sulfur dioxide  total sulfur dioxide  density    pH sulphates  alcohol  \
0                  45                 170.0   1.0010  3.00      0.45      8.8   
1                  14                 132.0   0.9940  3.30      0.49      9.5   
2                  30                  97.0   0.9951  3.26      0.44     10.1   
3                  47                 186.0   0.9956  3.19       0.4      9.9   
4                  47                 186.0   0.9956  3.19       0.4      9.9   
5                  30                  97.0   0.9951  3.26      0.44     10.1   
6                  30                 136.0   0.9949  3.18      0.47      9.6   

   quality  
0        6  
1        6  
2        6  
3        6  
4        6  
5        6  
6        6

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4898 entries, 0 to 4897
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         4898 non-null   float64
 1   volatile acidity      4898 non-null   float64
 2   citric acid           4898 non-null   float64
 3   residual sugar        4898 non-null   float64
 4   chlorides             4898 non-null   object 
 5   free sulfur dioxide   4898 non-null   object 
 6   total sulfur dioxide  4898 non-null   float64
 7   density               4898 non-null   float64
 8   pH                    4898 non-null   float64
 9   sulphates             4898 non-null   object 
 10  alcohol               4895 non-null   float64
 11  quality               4898 non-null   int64  
dtypes: float64(8), int64(1), object(3)
memory usage: 459.3+ KB


#Tutaj sb printuję te dane by zobaczyć, czy jest wszystko ok

 0   fixed acidity         4898 non-null   float64 - Stała Kwasowość
 1   volatile acidity      4898 non-null   float64 - Kwasowośc lotna
 2   citric acid           4898 non-null   float64 - Kwas cytrynowy
 3   residual sugar        4898 non-null   float64 - Cukier resztkowy
 4   chlorides             4898 non-null   object - Chlorki
 5   free sulfur dioxide   4898 non-null   object - Wolny dwutlenek siarki
 6   total sulfur dioxide  4898 non-null   float64 - Całkowity dwutlenek siarki
 7   density               4898 non-null   float64 - Gęstośc
 8   pH                    4898 non-null   float64 - pH
 9   sulphates             4898 non-null   object  - Siarczany
 10  alcohol               4895 non-null   float64 - Alkohol
 11  quality               4898 non-null   int64 - Jakość (ocena 0-10)

In [6]:
print(df['fixed acidity'].tolist())

[7.0, 6.3, 8.1, 7.2, 7.2, 8.1, 6.2, 7.0, 6.3, 8.1, 8.1, 8.6, 7.9, 6.6, 8.3, 6.6, 6.3, 6.2, 7.4, 6.5, 6.2, 6.4, 6.8, 7.6, 6.6, 7.0, 6.9, 7.0, 7.4, 7.2, 8.5, 8.3, 7.4, 6.2, 5.8, 7.3, 6.5, 7.0, 7.3, 7.3, 6.7, 6.7, 7.0, 6.6, 6.7, 7.4, 6.2, 6.2, 7.0, 6.9, 7.2, 6.6, 6.2, 6.4, 6.8, 6.9, 7.2, 6.0, 6.6, 7.4, 6.8, 6.0, 7.0, 6.6, 7.2, 6.2, 6.4, 6.7, 6.7, 7.4, 6.2, 6.8, 6.0, 8.6, 6.7, 7.4, 7.1, 7.0, 7.4, 6.7, 6.8, 7.1, 7.1, 7.3, 7.1, 7.1, 7.2, 6.8, 7.1, 7.1, 7.2, 6.1, 6.9, 6.9, 7.5, 7.1, 6.0, 8.6, 9.8, 6.0, 7.4, 7.1, 6.0, 7.5, 7.4, 7.3, 7.1, 7.1, 7.1, 6.9, 6.5, 7.2, 7.2, 6.7, 6.7, 5.5, 6.0, 7.0, 7.2, 7.3, 5.9, 7.8, 7.4, 6.9, 6.4, 6.7, 7.4, 6.5, 6.1, 6.1, 5.7, 6.5, 6.5, 6.6, 6.8, 6.7, 8.2, 7.1, 6.8, 8.1, 6.3, 7.1, 7.9, 8.5, 8.1, 6.3, 5.6, 6.4, 6.3, 6.9, 7.9, 7.6, 6.9, 7.2, 8.2, 7.4, 7.4, 6.1, 5.2, 5.2, 6.1, 6.4, 6.3, 7.4, 6.7, 7.0, 8.4, 6.7, 7.5, 9.8, 7.1, 7.9, 7.6, 6.3, 7.5, 6.9, 7.2, 5.9, 6.0, 6.4, 6.4, 7.1, 6.8, 6.9, 6.8, 6.8, 6.7, 5.9, 6.8, 6.5, 6.6, 6.8, 6.8, 6.6, 6.4, 6.4, 6.4, 6.8, 7.1, 6.8,

In [7]:
df['fixed acidity'].describe()

count    4898.000000
mean        6.854788
std         0.843868
min         3.800000
25%         6.300000
50%         6.800000
75%         7.300000
max        14.200000
Name: fixed acidity, dtype: float64

In [8]:
print(df['volatile acidity'].tolist())

[0.27, 0.3, 0.28, 0.23, 0.23, 0.28, 0.32, 0.27, 0.3, 0.22, 0.27, 0.23, 0.18, 0.16, 0.42, 0.17, 0.48, 0.66, 0.34, 0.31, 0.66, 0.31, 0.26, 0.67, 0.27, 0.25, 0.24, 0.28, 0.27, 0.32, 0.24, 0.14, 0.25, 0.12, 0.27, 0.28, 0.39, 0.33, 0.24, 0.24, 0.23, 0.24, 0.31, 0.24, 0.23, 0.18, 0.45, 0.46, 0.31, 0.19, 0.19, 0.25, 0.16, 0.18, 0.2, 0.25, 0.21, 0.19, 0.38, 0.2, 0.22, 0.19, 0.47, 0.38, 0.24, 0.35, 0.26, 0.25, 0.23, 0.24, 0.27, 0.3, 0.27, 0.23, 0.23, 0.24, 0.18, 0.32, 0.18, 0.54, 0.22, 0.2, 0.34, 0.22, 0.43, 0.44, 0.39, 0.25, 0.43, 0.44, 0.39, 0.27, 0.24, 0.21, 0.17, 0.26, 0.34, 0.265, 0.36, 0.34, 0.25, 0.12, 0.21, 0.305, 0.25, 0.13, 0.12, 0.23, 0.23, 0.33, 0.17, 0.27, 0.31, 0.41, 0.41, 0.485, 0.31, 0.14, 0.31, 0.32, 0.36, 0.24, 0.16, 0.19, 0.13, 0.19, 0.39, 0.24, 0.3, 0.3, 0.26, 0.24, 0.425, 0.24, 0.27, 0.27, 0.23, 0.37, 0.19, 0.28, 0.31, 0.37, 0.21, 0.21, 0.2, 0.255, 0.16, 0.595, 0.34, 0.25, 0.22, 0.18, 0.25, 0.18, 0.23, 0.24, 0.24, 0.32, 0.44, 0.44, 0.32, 0.22, 0.36, 0.24, 0.24, 0.23, 0.27, 

In [12]:
df['volatile acidity'].describe()
print(df['volatile acidity'].isna().sum())

0
